File: Sorting/binarysearch.c
#include<stdio.h>
#include<stdbool.h>
void bsearch(int arr[],int target);
int main(){
    int n=4;
    int arr[]={3,4,7,8};
    bsearch(arr,4);
}
void bsearch(int arr[],int target){
    int n=5;
    int si=0;
    int ei=n-1;
    while(si<=ei){
        int mid=(si+ei)/2;
        if(arr[mid]==target){
            printf(" %d is found at index %d",arr[mid],mid);
            break;
        }
        else if(arr[mid]< target){
            si=mid+1;
        }
        else{
            ei=mid-1;
        }
    }

    
}

File: 1-array/BinarySearch.java
import java.util.*;
public class BinarySearch {
    public static int search(int arr[],int target){
        int start=0;
        int end=arr.length-1;
        while(start<=end){
            int mid=(start+end)/2;
            if(arr[mid]==target){
                System.out.println("found");
                return target;
            }
            else if(arr[mid]<target){
                start=mid+1;
            }
            else{
                end=mid-1;
            }


        }
        return -1;

    }

    public static void main(String args[]){
        int arr[]={1,2,3,5,6};
        System.out.println(search(arr,5));
       

    }
    
}
